decoder.go

```diff
@@ -84,13 +84,84 @@ func (d *Decoder) Decode(dst interface{}, src map[string][]string) error {
 			errors[path] = UnknownKeyError{Key: path}
 		}
 	}
+	errors.merge(d.setDefaults(t, v))
 	errors.merge(d.checkRequired(t, src))
 	if len(errors) > 0 {
 		return errors
 	}
 	return nil
 }
 
+//setDefaults sets the default values when the `default` tag is specified,
+//default is supported on basic/primitive types and their pointers,
+//nested structs can also have default tags
+func (d *Decoder) setDefaults(t reflect.Type, v reflect.Value) MultiError {
+	struc := d.cache.get(t)
+	if struc == nil {
+		// unexpect, cache.get never return nil
+		return MultiError{"default-" + t.Name(): errors.New("cache fail")}
+	}
```

Comment:
Noting that I agree that cache.get() can never get to this position based on it's current implementation.